src/Platform/Microsoft.Testing.Platform/Messages/PropertyBag.cs

```diff
@@ -230,11 +230,11 @@ public TProperty Single<TProperty>()
             throw new InvalidOperationException($"Could not find a property of type '{typeof(TProperty)}'.");
         }
 
-        IEnumerable<TProperty> matchingValues = _property is null ? Array.Empty<TProperty>() : _property.OfType<TProperty>();
+        TProperty[] matchingValues = _property is null ? [] : _property.OfType<TProperty>().ToArray();
```

Comment:
Can we have a benchmark for this? I'd personally avoid changing it without a benchmark showing clear benefit.